### ISSUE TITLE
maybe some bugs?

### ISSUE THREAD

**Issue by Dinngger** [Jittor/JNeRF#77]
1. In `grid_encode.py`, `log2` in Python is not exactly same with `std::log2` in C++. This can lead to a mismatch in `scale` in some cases.
2. The `jt.nn.softplus` is not the same as the softplus in PyTorch. I don't know if this will cause any problems.
3. In GridEncode, JNeRF implementation doesn't have the loss scale while tcnn has it (with loss_scale=128). This will cause gradient disappearance.
4. When computing `alpha` in sdf, JNeRF uses safe_clip(0.0, 1.0). I think clamp_(0.0, 1.0) should be used.
5. The `cumprod` function in jittor is unsafe! Have a look at PyTorch implementation or mine in Jeuralangelo.
6. JNeRF doesn't have the weight norm but I found it in Jittor's source. I don't know why.

**Comment by qianqiuzzz** [comment]
I met No.3 error，loss increases sharply during training. do you have any solution to fix it?

**Comment by Dinngger** [comment]
> I met No.3 error，loss increases sharply during training. do you have any solution to fix it?

refer to my [code](https://github.com/Dinngger/Jeuralangelo/blob/98ee26c11ad5392bc040b133a33c1f38e5e690bd/hash_encoder/grid_encode.py#L120), it might help you.
